### ISSUE TITLE
Support drive item copy and progress watching in v5.0

### ISSUE THREAD

**Issue by olivermue** [microsoftgraph/msgraph-sdk-dotnet#1709]
**Is your feature request related to a problem? Please describe.**
The v5 SDK supports copying of a file in OneDrive with this call:
```csharp
var drive = await service.Users[sourceId].Drive.GetAsync();
var result = await service.Drives[drive.Id].Items[sourceItem.Id].Copy.PostAsync(new CopyPostRequestBody
{
    Name = sourceItem.Name,
    ParentReference = reference,
});
```
Unfortunately the response of this call isn't a `DriveItem` as the SDK says. According to [the documenation](https://learn.microsoft.com/en-us/graph/api/driveitem-copy?view=graph-rest-1.0#response-1) it just returns in the response header a location url where the copy progress can be monitored and described [here](https://learn.microsoft.com/en-us/graph/long-running-actions-overview).

This whole mechanism still seems to be unsupported from the SDK, but was already mentioned in #1566 and #226.

**Describe the solution you'd like**
The copy command should return something that can be fed into another helper class that is able to make the whole process of monitoring the copy progress, similiar to the helper class `PageIterator<T>` that wraps the needed stuff to iterate through larger collections.

**Describe alternatives you've considered**
In the v4 SDK everything was based on `HttpRequestMessage` which in such cases could be retrieved with `.Request()`, then manipulated and sent through a `HttpClient`. In v5 the new abstraction layer is Kiota and the biggest problem seems to be to access the response header to build this mechanism by ourselves, because getting the request can be done with `.ToRequestInformation()` and then we can manipulate it as needed. Sending the modified request can be done with `graphServiceClient.RequestAdapter.SendAsync()`, but we won't be able to read and parse the response header from that call. So until this stuff is directly available in the SDK, please describe how to read the raw response (including the header) with the `IRequestAdapter`.

**Comment by andrueastman** [comment]
Thanks for raising this @olivermue,

At the moment, it should be possible to read the Location header using the [NativeResponseHandler](https://github.com/microsoft/kiota-abstractions-dotnet/blob/main/src/NativeResponseHandler.cs) to override the default responseHandler to get the raw native response as below. 

```cs
var nativeResponseHandler = new NativeResponseHandler();
await graphServiceClient.Drives["driveId"].Items["sourceItemId"].Copy.PostAsync(new CopyPostRequestBody
                                                                    {
                                                                        Name = sourceItem.Name,
                                                                        ParentReference = reference,
                                                                    }, 
                                                                    requestConfiguration => requestConfiguration.Options.Add(new ResponseHandlerOption(){ ResponseHandler = nativeResponseHandler }));

var responseMessage = nativeResponseHandler.Value as HttpResponseMessage;
if (responseMessage.IsSuccessStatusCode) 
{
    var locationHeader = responseMessage.Headers.Location;
}
```

**Comment by andrueastman** [comment]
@olivermue 

I believe it may be possible to use the AsyncMonitor in this scenario as below.

```cs
  var locationHeader = responseMessage.Headers.Location;
  var asyncMonitor = new AsyncMonitor<DriveItem>(graphServiceClient, locationHeader.OriginalString);
  var progress = new Progress<AsyncOperationStatus>(status => 
  {
      Console.WriteLine($"Status: {status.Status}");
      Console.WriteLine($"Percentage complete: {status.PercentageComplete}");
      Console.WriteLine($"Operation: {status.Operation}");
  });
  var result = await asyncMonitor.PollForOperationCompletionAsync(progress, CancellationToken.None);
```

**Comment by olivermue** [comment]
Both worked. 🥳
In the past we already used the `AsyncMonitor<T>`. I just missed it was also ported to Kiota and at the point I was currently working on something similar to `AsyncMonitor<>` was implemented by ourself. The workaround to access the native response header is also great, but I think, I never found that on my own way. So would be great if this kind of hack could be documented somewhere (maybe in v5.0 upgrade guide?) and in the long run it would be perfect, if the SDK would correctly return something from `Copy.PostAsync()` that can directly be fed into the `AsyncMonitor<>`.
